common/djangoapps/student/tests/test_reset_password.py

```diff
@@ -281,6 +282,24 @@ def test_password_reset_fail(self):
         self.assertEqual(resp.status_code, 200)
         self.assertFalse(User.objects.get(pk=self.user.pk).is_active)
 
+    def test_password_reset_prevent_auth_user_writes(self):
+        with waffle().override(PREVENT_AUTH_USER_WRITES, True):
+            url = reverse(
+                "password_reset_confirm",
+                kwargs={"uidb36": self.uidb36, "token": self.token}
+            )
+            good_reset_req = self.request_factory.get(url)
+            response = password_reset_confirm_wrapper(good_reset_req, self.uidb36, self.token)
+            assert response.context_data['err_msg'] == SYSTEM_MAINTENANCE_MSG
+            self.user.refresh_from_db()
+            assert not self.user.is_active
+
+            good_reset_req = self.request_factory.post(url)
+            response = password_reset_confirm_wrapper(good_reset_req, self.uidb36, self.token)
+            assert response.context_data['err_msg'] == SYSTEM_MAINTENANCE_MSG
+            self.user.refresh_from_db()
+            assert not self.user.is_active
```

Comment:
Nit: Maybe make a for loop with the two methods in an array to make the test more clear that the same behavior is expected for GET * POST?

Reply:
Sure, refactoring.